base/strings/string.jl

```diff
@@ -34,46 +34,35 @@ const utf8_trailing = [
 ## required core functionality ##
 
 function endof(s::String)
-    d = s.data
-    i = length(d)
-    @inbounds while i > 0 && is_valid_continuation(d[i])
+    p = pointer(s)
```

Comment:
We need a `gcroot` intrinsic with a semantic similar to `ccall` to make this usage gc safe.